### DISCUSSION TITLE
Areablock getCurrent return index - 1

### DISCUSSION THREAD

**Discussion by sasori-developer** [pimcore/pimcore#8014] [Q&A / Support]
Editable of type Areablock has method getCurrent that should return the current iterator step and getCurrentIndex that should return the current index of the editable.
If i use those method I don't get the correct current index of my areablock.

Testing with an array of areablock i got current -1, 0 and 1 and so the currentIndex doesn't correctly match the elements.

To avoid this behaviours I've reused the properties and didn't subtract 1 from the iterator index.

This is the original implementation:
`return $this->indices[$this->getCurrent()]['key'];`

I've reimplemented like this:
`return $this->indices[$this->current]['key'];`

And now it correctly works.

Is it an expected behaviour? Should I open a PR to fix it?

**Comment by mittererr** [comment]
Same problem here (Pimcore 10.0.2)! PR?
